### PR TITLE
add logic to auto subscribe cwlgroups to firehose

### PR DESCRIPTION
**Pull request by dgwhited**
## Description:
Adds an event trigger for new cloudwatch log groups with the /aws/eks prefix, a step function to subscribe that log group to a kinesis firehose that dumps into the cms cloud logging bucket.

### Submitter Checklist
-  [x] Is there an impact on Auditing and Logging procedures or capabilities?
-  [ ] Is there an impact on Authentication procedures or capabilities?
-  [ ] Is there an impact on Authorization procedures or capabilities?
-  [ ] Is there an impact on Communication Security procedures or capabilities?
-  [ ] Is there an impact on Cryptography procedures or capabilities?
-  [ ] Is there an impact on Sensitive Data procedures or capabilities?
-  [ ] Is there an impact on any other security-related procedures or capabilities?
-  [ ] No security impacts identified.

Security Risks Identified - For any applicable items on the "Submitter Checklist," describe the impact of the change and any implemented mitigations.

Sends /aws/eks/ logs to the s3 bucket for SDL ingestion.

### CODE REVIEW

File: cwl_subscriptions/iam.tf [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    ]
+  })
+
+  inline_policy {
+    name = "panther-cloudwatch-firehose-policy"
+    policy = jsonencode({
+      "Version" : "2012-10-17"
+      "Statement" : [
+        {
+          "Action" : "firehose:*"
```
Comment (dgwhited) [inline cwl_subscriptions/iam.tf]: i need to fix this

File: cwl_subscriptions/events.tf [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      ]
+    },
+    "errorCode": [
+      { 
+        "exists": false
+      }
+    ]
+  }
+}
+EOF
```
Comment (bushong1) [inline cwl_subscriptions/events.tf]: Avoid in-line json
```suggestion
  event_pattern = jsonencode({
    detail = {
      errorCode = [
        {
          exists = false
        },
      ]
      eventName = [
        "CreateLogGroup",
      ]
      eventSource = [
        "logs.amazonaws.com",
      ]
      requestParameters = {
        logGroupName = [
          {
            prefix = "/aws/eks/"
          },
        ]
      }
    }
    detail-type = [
      "AWS API Call via CloudTrail",
    ]
    source = [
      "aws.logs",
    ]
  })
```

File: cwl_subscriptions/iam.tf [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+}
+
+# Step Function Role
+resource "aws_iam_role" "sfn_new_loggroup" {
+  name                 = "new-loggroup-sfn"
+  path                 = var.iam_role_path
+  permissions_boundary = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:policy/cms-cloud-admin/developer-boundary-policy"
+
+  inline_policy {
+    name   = "new-loggroup-sfn"
```
Comment (bushong1) [inline cwl_subscriptions/iam.tf]: new log group seems to suggest the policy allows creating new log groups, but i'm not seeing that in the policy doc below

File: cwl_subscriptions/iam.tf [hunk 4]
```diff
@@ -0,0 +1,164 @@
+# firehose to s3 role
+resource "aws_iam_role" "firehose_s3_role" {
+  name                 = "${var.cluster_name}-FirehoseS3Role"
+  path                 = var.iam_role_path
+  permissions_boundary = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:policy/cms-cloud-admin/developer-boundary-policy"
```
Comment (bushong1) [inline cwl_subscriptions/iam.tf]: Should the permissions_boundary suffix be something we pass in as a var?  It's how we do it everywhere else. Or will this be standalone and not deployed with terragrunt?

Comment (dgwhited) [reply cwl_subscriptions/iam.tf]: yeah thats fine, then we just need to build the arn in tg insead

**Review by bushong1 — APPROVED** [review]

**Pull request merged**